replicator/src/core.rs

```diff
@@ -102,11 +105,16 @@ pub async fn start_replicator() -> anyhow::Result<()> {
     Ok(())
 }
 
-async fn init_state_store(
-    config: &ReplicatorConfig,
-) -> anyhow::Result<impl StateStore + Clone + Send + Sync + fmt::Debug + 'static> {
+async fn init_state_store(config: &ReplicatorConfig) -> anyhow::Result<BoxedStateStore> {
     match config.state_store {
-        StateStoreConfig::Memory => Ok(MemoryStateStore::new()),
+        StateStoreConfig::Memory => Ok(BoxedStateStore::new(MemoryStateStore::new())),
```

Comment:
For our case, I am struggling to see the benefit of dynamic dispatch via `BoxedStateStore`. Would we ever want to swap the state store implementation at runtime?